Samples/HelloWorld/OrleansClient/Program.cs

```diff
@@ -26,29 +45,22 @@ static int Main(string[] args)
                     Console.WriteLine("Client successfully connect to silo host");
                     break;
                 }
-                catch (SiloUnavailableException e)
+                catch (SiloUnavailableException)
                 {
-                    attemp++;
-                    if (attemp > initializeAttemptsBeforeFailing)
+                    attempt++;
+                    Console.WriteLine($"Attempt {attempt} of {initializeAttemptsBeforeFailing} failed to initialize the Orleans client.");
+                    if (attempt > initializeAttemptsBeforeFailing)
                     {
-                        Console.WriteLine("Connect to silo host failed due to " + e);
-                        Console.WriteLine("Press Enter to terminate...");
-                        Console.ReadLine();
-                        return 1;
+                        throw;
```

Comment:
You just throw the exception here without printing it to log or cml. would the user know why the client program exited ?